datafusion/src/physical_plan/functions.rs

```diff
@@ -1373,20 +1370,26 @@ impl PhysicalExpr for ScalarFunctionExpr {
     }
 
     fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
-        // evaluate the arguments
-        let inputs = self
-            .args
-            .iter()
-            .map(|e| e.evaluate(batch))
-            .collect::<Result<Vec<_>>>()?;
+        // evaluate the arguments, if there are no arguments we'll instead pass in a null array of
+        // batch size (as a convention)
+        let inputs = match self.args.len() {
+            0 => vec![ColumnarValue::Array(Arc::new(NullArray::new(
```

Comment:
I guess then another option is to add a new type of node which models no arg functions

Reply:
See also https://github.com/apache/arrow-datafusion/pull/307

Reply:
Good call @jorgecarleitao - I feel both approaches are a bit "workarounds" to the issue of not having something in design for zero-arguments functions.

I feel what would be "cleaner":
* passing a `usize` argument for the batch size (`so you don't have to do `args[0].num_rows()` or something like that but use the length anywhere. This might break some stuff(?)
* having another node for zero-argument functions. This might duplicate some stuff(?)

Reply:
@alamb @Dandandan and @jorgecarleitao thanks for the discussion.

I agree that both ways are similar and equally “hacky” but for lack of a better solution they are okay. I'd slightly prefer null array because there's no ScalarValue::USize and having to convert from/to UInt32 is a bit cumbersome.

Reply:
@jorgecarleitao  / @Dandandan  / @Jimexist  are we all cool with using a `NullArray` to pass "how many rows are passed to this function that has no input arguments"?

Reply:
Yeah, let's do that for now

Reply:
👍

Reply:
So I think the plan should be merge  https://github.com/apache/arrow-datafusion/pull/328 and then rebase this PR to pick up that change